### PR TITLE
News: Allow users to request announcements

### PR DESCRIPTION
**Pull request by DeathlyPlays**

### CODE REVIEW

**Comment by Lord-Haji** [comment]
Use nef not fs  on a json file

File: spacialgaze-plugins/news.js [hunk 1]
```diff
@@ -93,11 +110,36 @@ exports.commands = {
...
+			if (!user.named) return this.errorReply('You must have a name before requesting an announcement.');
+			if (!this.canTalk()) return this.errorReply("You can't use this command while unable to speak.");
+			if (!target) return this.sendReply("/news request [message] - Requests a news announcement from SG Staff.");
+			if (target.length < 1) return this.sendReply("/news request [message] - Requests a news announcement from SG Staff.");
+			let newsId = (Object.keys(newsRequests).length + 1);
+			let d = new Date();
+			let MonthNames = ["January", "February", "March", "April", "May", "June",
+				"July", "August", "September", "October", "November", "December",
+			];
+			console.log(newsId);
```
Comment (Lord-Haji) [inline spacialgaze-plugins/news.js]: do not log this

**Review by Lord-Haji — CHANGES_REQUESTED** [review]
Coding not up to the mark this seems like scratched off from ``requesthelp`` you need to redo it totally

File: spacialgaze-plugins/news.js [hunk 2]
```diff
@@ -93,11 +110,36 @@ exports.commands = {
...
 			this.popupReply("|wide||html|You will no longer automatically receive SpacialGaze News.<br><hr><center><button class='button' name='send' value='/news'>View News</button></center>");
 		},
+		request: function (target, room, user) {
+			if (!user.named) return this.errorReply('You must have a name before requesting an announcement.');
+			if (!this.canTalk()) return this.errorReply("You can't use this command while unable to speak.");
+			if (!target) return this.sendReply("/news request [message] - Requests a news announcement from SG Staff.");
+			if (target.length < 1) return this.sendReply("/news request [message] - Requests a news announcement from SG Staff.");
+			let newsId = (Object.keys(newsRequests).length + 1);
+			let d = new Date();
+			let MonthNames = ["January", "February", "March", "April", "May", "June",
```
Comment (Lord-Haji) [inline spacialgaze-plugins/news.js:120]: make a single array and refer it wherever you need
this is a dup

**Comment by DeathlyPlays** [comment]
@Lord-Haji what would I use to make it nef then?  I don't use nef fs so I don't know how to convert it for you

File: spacialgaze-plugins/news.js (hunk 2 above)
Comment (DeathlyPlays) [reply spacialgaze-plugins/news.js:120]: the date thing?

Comment (Lord-Haji) [reply spacialgaze-plugins/news.js:120]: Yee

**Comment by Lord-Haji** [comment]
Check the documentation
https://www.npmjs.com/package/nef

**Comment by DeathlyPlays** [comment]
Probably unnecessary anyways lmao

**Comment by DeathlyPlays** [comment]
Too tired rn

**Comment by DeathlyPlays** [comment]
Gotta find a stupid issue when I updated to main

**Pull request closed without merging**